lib/Target/OpenFhePke/Interpreter.h

```diff
@@ -0,0 +1,169 @@
+#ifndef LIB_TARGET_OPENFHEPKE_INTERPRETER_H_
+#define LIB_TARGET_OPENFHEPKE_INTERPRETER_H_
+
+#include <string>
+#include <variant>
+#include <vector>
+
+#include "lib/Dialect/LWE/IR/LWEOps.h"
+#include "lib/Dialect/Openfhe/IR/OpenfheOps.h"
+#include "llvm/include/llvm/ADT/DenseMap.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Affine/IR/AffineOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"    // from @llvm-project
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"   // from @llvm-project
+#include "mlir/include/mlir/Dialect/SCF/IR/SCF.h"        // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tensor/IR/Tensor.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinOps.h"             // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"              // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"                  // from @llvm-project
+#include "src/pke/include/openfhe.h"                     // from @openfhe
+
+namespace mlir {
+namespace heir {
+namespace openfhe {
+
+using CCParamsT = lbcrypto::CCParams<lbcrypto::CryptoContextCKKSRNS>;
+using CiphertextT = lbcrypto::Ciphertext<lbcrypto::DCRTPoly>;
+using CryptoContextT = lbcrypto::CryptoContext<lbcrypto::DCRTPoly>;
+using EvalKeyT = lbcrypto::EvalKey<lbcrypto::DCRTPoly>;
+using PlaintextT = lbcrypto::Plaintext;
+using PrivateKeyT = lbcrypto::PrivateKey<lbcrypto::DCRTPoly>;
+using PublicKeyT = lbcrypto::PublicKey<lbcrypto::DCRTPoly>;
+using FastRotPrecompT = std::shared_ptr<std::vector<lbcrypto::DCRTPoly>>;
+
+struct TypedCppValue {
+  using Variant =
+      std::variant<std::monostate,                    // NULL_TY
+                   bool,                              // BOOL
+                   int,                               // INT
+                   float,                             // FLOAT
```

Comment:
do we also use floats in the emitter? maybe we should use doubles to handle 64-bit floating points?